common/primitives/src/offchain.rs

```diff
@@ -0,0 +1,77 @@
+use codec::{Decode, Encode};
+use sp_io::offchain_index;
+use sp_runtime::offchain::{
+	storage::{StorageRetrievalError, StorageValueRef},
+	storage_lock::{StorageLock, Time},
+	Duration,
+};
+use sp_std::fmt::Debug;
+
+const DB_LOCK: &[u8] = b"lock";
+
+const LOCK_DEADLINE: u64 = 1000;
+
```

Comment:
1 sec lock , too pessimistic?

Reply:
You could try setting it to 500ms.  Lowering the timeout value may increase the chances of lock contention, but it will also decrease the time that the node needs to wait for a lock, which will lead to a higher transaction throughput.

Reply:
I can't find the source but somewhere I read the default timeout was 100ms.

Reply:
I agree shorter the better, will see the outcome of discussion, there three types of locks

* Time : duration based
* Block: lock acquired on block
* BlockTime: hybrid of both 
https://paritytech.github.io/substrate/master/sp_runtime/offchain/storage_lock/index.html

For our case i believe timed lock is enough

Reply:
Found that default is even much larger, I think based on how typically of offchain_worker will take to write to db? I think we can go with much lower value
https://github.com/paritytech/substrate/blob/2eb8ad273eeabea380fce00c10893ac788b1cbde/primitives/runtime/src/offchain/storage_lock.rs#L74

Reply:
We can keep in mind that lock is acquired per msa_id so i believe its safe to use even default value, 1 sec sounds reasonable thinking about many many msa_ids being indexed over time